streamrip/client/downloadable.py

```diff
@@ -124,9 +124,14 @@ def __init__(self, session: aiohttp.ClientSession, info: dict):
         self.session = session
         self.url = info["url"]
         self.source: str = "deezer"
-        max_quality_available = max(
-            i for i, size in enumerate(info["quality_to_size"]) if size > 0
-        )
+        try:
+            max_quality_available = max(
+                i for i, size in enumerate(info["quality_to_size"]) if size > 0
+            )
+        except:
```

Comment:
Instead of doing a bare `except` it would be better if you check if `info['quality_to_size']` has length 0, and then throw the error

Reply:
I suspect by watching the code that `info["quality_to_size"]` could come with 3 0-items, as `[0, 0, 0]` for each quality type.
It could be confirmed with a debug log from the linked issue, cause I cannot reproduce it as the playlist/track link was not provided either.
Also not sure what happens if you just assign here `max_quality_available` to 0 and therefore `self._size` will be 0. Will it throw an exception when trying to download or will skip it? Or it could skip the track there anyway, maybe would be a bit better design...

Or even skip before this constructor is called, to keep it thin )

Reply:
Then maybe

```python
available_qualities = [s for s in info['quality_to_size'] if s > 0]
if len(available_qualities) == 0:
    ...
```